R-packages/evalcast/vignettes/intro-evalcast.Rmd

```diff
@@ -44,87 +48,142 @@ library(evalcast)
+predictions_cards_cmu %>% 
+  filter(geo_value == "mi", forecast_date == forecast_dates_july[1]) %>%
+  select(geo_value, quantile, value, forecast_date) %>% str()
 ```
 
+Forecasts are generally probabilistic, meaning that rather than predicting a single value, a distribution of values at different quantiles. For example, the Covidhub ensemble forecaster would have predicted the following distribution in California for `r predictions_cards_cmu$ahead[1]` week(s) after `r predictions_cards_cmu$forecast_date[1]`
+```{r}
+predictions_cards_ens %>%
+  filter(geo_value == "ca", forecast_date == forecast_dates_july[1]) %>%
+  select(quantile, value) %>% str()
+```
+
+An `NA` in the `quantile` column indicates a point forecast (possibly different from the median).
+
+
 ### Evaluate the performance of a forecaster
 
 Now that we've made these predictions, we'd like to know how well they perform.  By default, `evalcast` uses the following three performance measures, but these can be easily substituted:
 
 ```{r}
 err_measures <- list(wis = weighted_interval_score,
                      ae = absolute_error,
-                     coverage_80 = interval_coverage(alpha = 0.2))
+                     coverage_80 = interval_coverage(coverage = 0.8))
 ```
 
 We may now create scorecards for each forecaster's predictions or for a filtered subset. 
 
 ```{r}
-scorecards_yyg <- evaluate_predictions(filter_predictions(predictions_cards_yyg, ahead = 3), 
-                                   err_measures, 
-                                   backfill_buffer = 10)
-pcards_cmu_death3 <- filter_predictions(predictions_cards_cmu, 
-                                        ahead = 3,
-                                        response_data_source = "jhu-csse",
-                                        response_signal = "deaths_incidence_num")
-scorecards_cmu <- evaluate_predictions(pcards_cmu_death3, 
-                                   err_measures, 
-                                   backfill_buffer = 10)
-scorecards <- evaluate_predictions(filter_predictions(predictions_cards, ahead = 3), 
-                                   err_measures, 
-                                   backfill_buffer = 10)
-```
-
-Let's look at the `baseline_forecaster`'s scorecard for 3-epiweek ahead forecasts.
+scorecards <- evaluate_predictions(
+  bind_rows(predictions_cards, 
+            predictions_cards_cmu, 
+            predictions_cards_ens) %>%
+    # not always data for these locations
+    filter(! geo_value %in% c("us", "vi", "gu", "mp")),
+  err_measures = err_measures,
+  backfill_buffer = 10
+  )
+```
+
+
+`evaluate_predictions()` returns a long data frame with one column for each error measure added to the `predictions_cards`. Keeping all quantile predictions may be useful for plotting or other downstream analysis. For a more compact version, use `collapse_cards()` to return one row for each geo_value/ahead/forecast_date/forecaster combination.
```

Comment:
Very minor, but above you refer to it as a long `data.frame` and here it's a "data frame". Either one is fine, but I'd prefer to use the same style throughout.

Reply:
I may have called a `dataframe` too elsewhere...